subsys/bluetooth/controller/ll_sw/pdu.h

```diff
@@ -11,7 +11,7 @@
  * PDU fields sizes
  */
 
-#define PDU_PREAMBLE_SIZE(phy) (phy&0x3)
+#define PDU_PREAMBLE_SIZE(phy) ((((phy) & 0x3) >> 1) + 1)
```

Comment:
So both 0, and 1 is LE 1M Phy?
Is this macro supposed to accept enum values, or bit-field values?

This macro now returns 2 for values 2 and 3, which means that LE Coded PHY has a preamble of "2", that does not seem right to me. Is this macro supposed to support LE Coded PHY?

Reply:
@joerchan following definitions exist for phy (in hci.h, similar are in pdu.h)
#define BT_HCI_LE_EXT_SCAN_PHY_1M               BIT(0)
#define BT_HCI_LE_EXT_SCAN_PHY_2M               BIT(1)
#define BT_HCI_LE_EXT_SCAN_PHY_CODED            BIT(2)
The macro is supposed to accept bit-field values, and is only valid for 1M and 2M phy, not for Coded phy

See my other comment, IMO the proper fix is to ensure that the phy parameter is set to 1 in  case of the 1M phy being selected

Reply:
@kruithofa The reason why I ask is that there is a mix of PHY being an enum, or a bitfield.
For example:
```
#define BT_HCI_LE_PHY_1M                        0x01
#define BT_HCI_LE_PHY_2M                        0x02
#define BT_HCI_LE_PHY_CODED                     0x03
```
So here 1M and 2M are the same, but Coded is either 3 or 4.
Since it is `& 0x3` it looks like enum values, but I don't see LE Coded PHY handled by these macros.

Reply:
It's a good question. 
LE coded phy is not handled by this macro; for LE coded phy the preamble is 80uS, and it is handled in the PKT_DC_US macro  (pdu.h, line 109).
We'll have to discuss this in more detail with Vinayak

Reply:
AFAIK, control procedure use bitfield and hci use enum value. 0, BIT(0), BIT(1), and BIT(2) represent legacy (1M default), 1M preferred, 2M Preferred and Coded PHY preferred used in controller. Yes, we can discuss further when I can get back full time.

Reply:
HCI has both enums and bitfields. So it is easy to mess it up and select the wrong one.

Reply:
The value of 0 has to be considered as default (1M) PHY, setting PHY value to 1 is unnecessary overhead. HCI (enum) PHY values are derived using first-bit-set intrinsic function, there should not be mess up. That said, if there are occurrences of manual assignments, I will convert them to use intrinsic functions.

Reply:
Adding a right shift to the macro also introduce unnecessary overhead for non constant phy parameter.

Reply:
bit shift right and add instructions are 1 cycle each in ARM CM4, in comparison to compare plus branching taking 1 and 1 + P (pipeline refill) cycles, add 1 cycle to assign the value 1.

@kruithofa I have tried to search all assignments to `conn->lll.phy_tx` and cant find where 0 is assigned, could you give a try?

Reply:
@kruithofa could you make sometime to reproduce the incorrect timing value in #26252?

Reply:
We'll verify this using EBQ testing